apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -413,11 +415,76 @@ post_correct_tx(_Config) ->
             fee => Fee,
             nonce => Nonce}),
     {ok, SignedTx} = rpc(aec_keys, sign, [SpendTx]),
-    post_tx(SignedTx),
+    {ok, 200, _} = post_tx(base64:encode(aec_tx_sign:serialize_to_binary(SignedTx))),
     {ok, [SignedTx]} = rpc(aec_tx_pool, peek, [infinity]), % same tx 
     ok.
 
-%% TODO: POST broken tx (currently not handled in API Endpoint)
+post_broken_tx(_Config) ->
+    Amount = 7,
+    Fee = 2,
+    BlocksToMine = 10,
+    aecore_suite_utils:mine_blocks(aecore_suite_utils:node_name(?NODE),
+                                   BlocksToMine),
+    {ok, []} = rpc(aec_tx_pool, peek, [infinity]), % empty 
+    {ok, 200, _} = get_balance(), % account present
+    {ok, PubKey} = rpc(aec_keys, pubkey, []),
+    {ok, Nonce} = rpc(aec_next_nonce, pick_for_account, [PubKey]),
+    {ok, SpendTx} =
+        aec_spend_tx:new(
+          #{sender => PubKey,
+            recipient => random_hash(),
+            amount => Amount,
+            fee => Fee,
+            nonce => Nonce}),
+    {ok, SignedTx} = rpc(aec_keys, sign, [SpendTx]),
+    [T, V, SerializedTx, Sigs] = aec_tx_sign:serialize(SignedTx),
+    lists:foreach(
+        fun(Key) ->
+            BrokenTx = lists:filter(
+                fun(#{Key := _}) -> true;
+                (_) -> false
+                end,
+                SerializedTx),
+            EncodedBrokenTx = base64:encode(msgpack:pack([T, V, BrokenTx, Sigs])),
+            {ok, 400, #{<<"reason">> := <<"Invalid tx">>}} = post_tx(EncodedBrokenTx)
+        end,
+        [<<"type">>,
+         <<"vsn">>,
+         <<"sender">>,
+         <<"recipient">>,
+         <<"amount">>,
+         <<"fee">>,
+         <<"nonce">>]),
+
+    ok.
+
+post_broken_hash_tx(_Config) ->
+    Amount = 7,
+    Fee = 2,
+    BlocksToMine = 10,
+    aecore_suite_utils:mine_blocks(aecore_suite_utils:node_name(?NODE),
+                                   BlocksToMine),
+    {ok, []} = rpc(aec_tx_pool, peek, [infinity]), % empty 
+    {ok, 200, _} = get_balance(), % account present
+    {ok, PubKey} = rpc(aec_keys, pubkey, []),
+    {ok, Nonce} = rpc(aec_next_nonce, pick_for_account, [PubKey]),
+    NumberOfChecks = 10,
+    lists:foreach(
+        fun(_) ->
+            {ok, SpendTx} =
+                aec_spend_tx:new(
+                  #{sender => PubKey,
+                    recipient => random_hash(),
+                    amount => Amount,
+                    fee => Fee,
+                    nonce => Nonce}),
+            {ok, SignedTx} = rpc(aec_keys, sign, [SpendTx]),
+            <<_, BrokenHash/binary>> =
```

Comment:
I see apart from this line (that strips initial base64 char) is same as test case post_correct_tx, with new random fake recipient created each time.